daemon/daemon.go

```diff
@@ -335,6 +332,17 @@ configRetry:
 		exitWithCustomRC(configChangedRC, "Restarting to avoid leaking datastore connections")
 	}
 
+	if configParams.BPFEnabled {
+		// Check for BPF dataplane support before we do anything that relies on the flag being set one way or another.
+		if err := bpf.SupportsBPFDataplane(); err != nil {
+			log.Error("BPF dataplane mode enabled but not supported by the kernel.  Disabling BPF mode.")
+			_, err := configParams.OverrideParam("BPFEnabled", "false")
+			if err != nil {
+				log.WithError(err).Panic("Bug: failed to override config parameter")
+			}
+		}
+	}
```

Comment:
So IIUC the override is only computed when someone tries to enable BPF?  I imagined instead that the overrides would be computed independently (and unconditionally) at start of day.  What is your thinking on the preferred architecture here?

Reply:
I wondered about that, but it made it easier to only log the error in the "enabled but not supported case" if I did it this way.  Also felix restarts if the flag changes so we'll go through this code again if the flag is enabled while felix is running.

If I unconditionally set the override, felix wouldn't restart, but you'd also get no message saying that the feature was disabled so I could go either way.